io/kqueue.cpp

```diff
@@ -99,7 +99,7 @@ class KQueue : public MasterEventEngine, public CascadingEventEngine, public Res
             return 0;
         short ev = (interests == EVENT_READ) ? EVFILT_READ : EVFILT_WRITE;
         enqueue(fd, ev, EV_ADD | EV_ONESHOT, 0, CURRENT);
-        int ret = thread_usleep(timeout);
+        int ret = thread_usleep(timeout.timeout() ? timeout : Timeout(10));
```

Comment:
In case of timeout == 0, we'd better reap the events directly without blocking, so as to conform to the semantics.

Reply:
done